cloud/ovirt/ovirt_nics.py

```
@@ -0,0 +1,230 @@
+
+        # Locate the service that manages the virtual machines NICs:
+        vm_service = vms_service.vm_service(vm.id) 
+        nics_service = vm_service.nics_service()
+        vmnics_module = VmNicsModule(
+            connection=connection,
+            module=module,
+            service=nics_service,
+        )
+
+        # Find vNIC id of the network interface (if any):
+        profile = module.params.get('profile')
+        if profile and module.params['network']:
+            cluster_name = get_link_name(connection, vm.cluster)
+            dcs_service = connection.system_service().data_centers_service()
+            dc = dcs_service.list(search='Clusters.name=%s' % cluster_name)[0]
+            networks_service = dcs_service.service(dc.id).networks_service()
+            network = search_by_name(networks_service, module.params['network'])
+            for vnic in connection.system_service().vnic_profiles_service().list():
+                if vnic.name == profile and vnic.network.id == network.id:
+                    vmnics_module.vnic_id = vnic.id
+
+        # Handle appropriate action:
+        state = module.params['state']
+        if state == 'present':
+            ret = vmnics_module.create()
+        elif state == 'absent':
+            ret = vmnics_module.remove()
+        elif state == 'plugged':
+            vmnics_module.create()
+            ret = vmnics_module.action(
+                action='activate',
+                action_condition=lambda nic: not nic.plugged,
+                wait_condition=lambda nic: nic.plugged,
+            )
+        elif state == 'unplugged':
+            vmnics_module.create()
+            ret = vmnics_module.action(
+                action='deactivate',
+                action_condition=lambda nic: nic.plugged,
+                wait_condition=lambda nic: not nic.plugged,
+            )
+
+        module.exit_json(**ret)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Same here, please surface the full exception.